crypto/s2n_rsa_pss.c

```diff
@@ -176,14 +176,18 @@ static int s2n_rsa_pss_keys_match(const struct s2n_pkey *pub, const struct s2n_p
 
 static int s2n_rsa_pss_key_free(struct s2n_pkey *pkey)
 {
-    /* This object does not own the reference to the key --
-     * s2n_pkey handles it. */
+    struct s2n_rsa_key *rsa_key = &pkey->key.rsa_key;
```

Comment:
Missing `POSIX_ENSURE_REF` on `pkey` and `rsa_key`?

Reply:
https://github.com/aws/s2n-tls/pull/3473/commits/e3fa79b5cc3cf581cc4140baa47d28d19ca5983c

Reply:
ahh my bad. Seems like calls to `s2n_rsa_pss_key_free` can have `pkey->rsa_key == NULL`. This allows for multiple calls to free

Reply:
If `struct s2n_rsa_key rsa_key = &pkey->rsa_key` could be null then couldn't line 181 hit a null ptr exception?

Reply:
yes you are right `pkey->rsa_key` is not a valid field and that was causing the errors.

addressed in: https://github.com/aws/s2n-tls/pull/3473/commits/80a99901b591b1098413ade20ade0199a0cfd9dc